### PR TITLE
Add string benchmarks for replacing spaces and dasherizing

### PR DESCRIPTION
**Pull request by alexbaumhoer**
Let me know if you think there should be any changes.

### CODE REVIEW

File: spec/string_spec.rb [hunk 1]
```diff
@@ -29,4 +31,35 @@
     end
   end
 
+  it "replaces spaces with underscores" do
+    Benchmark.ips do |x|
+      str = Random.new.bytes(1024)
```
Comment (alexbaumhoer) [inline spec/string_spec.rb]: Will a random string like this be sufficient? I wasn't sure if I should be testing with a string that is guaranteed to have the characters we are testing replacement on?

Comment (onyxraven) [inline spec/string_spec.rb]: yeah you'd probably want to have some of those characters.

I wonder if Faker is a better choice for this

Comment (onyxraven) [inline spec/string_spec.rb]: I'd merge this and the below since they're all tested right?

Comment (alexbaumhoer) [inline spec/string_spec.rb]: Hmmmm. How would you imagine using Faker? There's not a built in space or underscore helper I imagine. Build your own "Faker" class?

**Comment by alexbaumhoer** [comment]
Whoops. They are all tested. See the results below:

```
String
Warming up --------------------------------------
        tr(' ', '_')    38.003k i/100ms
      gsub(' ', '_')    22.221k i/100ms
Calculating -------------------------------------
        tr(' ', '_')    521.077k (± 3.8%) i/s -      2.622M
      gsub(' ', '_')    251.420k (±10.6%) i/s -      1.244M
  replaces spaces with underscores

Warming up --------------------------------------
           dasherize    17.609k i/100ms
        tr('_', '-')    38.806k i/100ms
      gsub('_', '-')    19.719k i/100ms
Calculating -------------------------------------
           dasherize    200.666k (± 3.1%) i/s -      1.004M
        tr('_', '-')    498.991k (± 3.4%) i/s -      2.522M
      gsub('_', '-')    228.615k (± 3.4%) i/s -      1.144M
  dasherizes
```

File: spec/string_spec.rb (hunk 1 above)
Comment (onyxraven) [inline spec/string_spec.rb]: Faker::Internet.user_name('Nancy Johnson', %w(. _ -)) #=> "johnson-nancy" looks like a good candidate

Comment (onyxraven) [inline spec/string_spec.rb]: but, what you're looking for is potential spaces, so anything that generates multiple words.

**Comment by alexbaumhoer** [comment]
Latest test results with Faker generating our strings:

```
String
Warming up --------------------------------------
           dasherize    71.705k i/100ms
        tr('_', '-')    90.300k i/100ms
      gsub('_', '-')    65.761k i/100ms
Calculating -------------------------------------
           dasherize      1.350M (± 7.6%) i/s -      6.740M
        tr('_', '-')      2.027M (±10.1%) i/s -     10.023M
      gsub('_', '-')      1.189M (± 9.0%) i/s -      5.918M
  dasherizes
Warming up --------------------------------------
        tr(' ', '_')    84.253k i/100ms
      gsub(' ', '_')    34.474k i/100ms
Calculating -------------------------------------
        tr(' ', '_')      1.450M (± 7.9%) i/s -      7.246M
      gsub(' ', '_')    438.080k (± 5.5%) i/s -      2.206M
  replaces spaces with underscores
```

It looks like ```tr``` is killing it?

**Comment by onyxraven** [comment]
ya, TR wins, bigtime.

**Comment by onyxraven** [comment]
Thanks!  you done with this branch?  I'll merge it

**Comment by alexbaumhoer** [comment]
Yep, all good!

**Pull request merged**